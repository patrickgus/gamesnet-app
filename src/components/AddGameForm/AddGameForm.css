.AddGameForm {
  display: flex;
  flex-direction: column;
}

.AddGameForm label {
  color: rgb(0, 224, 224);
}

.AddGameForm input,
textarea {
  opacity: 0.8;
}

.AddGameForm__buttons {
  display: flex;
  justify-content: space-evenly;
}
