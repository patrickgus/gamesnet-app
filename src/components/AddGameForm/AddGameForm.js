import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import GameListContext from "../../contexts/GameContext";
import GameApiService from "../../services/game-api-service";
import {
  Input,
  Required,
  Textarea,
  Button
} from "../../components/Utils/Utils";

class AddGameForm extends Component {
  static contextType = GameListContext;

  handleSumbit = e => {
    e.preventDefault();
    const { title, description, rated, platforms } = e.target;

    GameApiService.postGame(
      title.value,
      description.value,
      rated.value,
      platforms.value
    )
      .then(this.context.addGame)
      .then(() => {
        title.value = "";
        description.value = "";
        rated.value = "";
        platforms.value = "";
        this.props.history.push("/games");
      })
      .catch(this.context.setError);
  };

  handleCancel = e => {
    e.preventDefault();
    this.props.history.push("/games");
  };

  render() {
    return (
      <form className="AddGameForm" onSubmit={this.handleSumbit}>
        <div className="game_title">
          <label htmlFor="AddGameForm__game_title">
            Game title <Required />
          </label>
          <Input
            name="title"
            type="text"
            required
            id="AddGameForm__game_title"
          />
        </div>
        <div className="game_cover">
          <label htmlFor="AddGameForm__game_cover">Game cover image url</label>
          <Input name="game_cover" type="text" id="AddGameForm__game_cover" />
        </div>
        <div className="game_description">
          <label htmlFor="AddGameForm__game_description">
            Game description <Required />
          </label>
          <Textarea
            required
            aria-label="Write a brief description..."
            name="description"
            id="AddGameForm__game_description"
            rows="15"
            placeholder="Write a brief description..."
          />
        </div>
        <div className="rated">
          <label htmlFor="AddGameForm__rated">
            Rated <Required />
          </label>
          <Input
            name="rated"
            type="text"
            required
            id="AddGameForm__rated"
            placeholder="E, T, M, etc."
          />
        </div>
        <div className="platforms">
          <label htmlFor="AddGameForm__platforms">
            Platforms <Required />
          </label>
          <Input
            name="platforms"
            type="text"
            required
            id="AddGameForm__platforms"
            placeholder="Playstation, Xbox, PC, etc."
          />
        </div>
        <Button type="submit">Add Game</Button>
        <Button onClick={this.handleCancel}>Cancel</Button>
      </form>
    );
  }
}

export default withRouter(AddGameForm);
