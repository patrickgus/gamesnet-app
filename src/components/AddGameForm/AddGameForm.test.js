import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import AddGameForm from "./AddGameForm";

describe("AddGameForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddGameForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer.create(<AddGameForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
