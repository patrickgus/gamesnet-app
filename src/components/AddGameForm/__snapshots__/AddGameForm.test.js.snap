// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddGameForm component renders the form as expected 1`] = `
<form
  className="AddGameForm"
  onSubmit={[Function]}
>
  <div
    className="AddGameForm__title"
  >
    <label
      htmlFor="title"
    >
      Game title 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="AddGameForm__description"
  >
    <label
      htmlFor="description"
    >
      Game description 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <textarea
      aria-label="Write a brief description..."
      className="Textarea "
      id="description"
      name="description"
      placeholder="Write a brief description..."
      required={true}
      rows="15"
    />
  </div>
  <div
    className="AddGameForm__rated"
  >
    <label
      htmlFor="rated"
    >
      Rated 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="rated"
      name="rated"
      placeholder="E, T, M, etc."
      required={true}
      type="text"
    />
  </div>
  <div
    className="AddGameForm__platforms"
  >
    <label
      htmlFor="platforms"
    >
      Platforms 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="platforms"
      name="platforms"
      placeholder="Playstation, Xbox, PC, etc."
      required={true}
      type="text"
    />
  </div>
  <div
    className="AddGameForm__buttons"
  >
    <button
      className="Button "
      type="submit"
    >
      Add Game
    </button>
    <button
      className="Button "
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
